kornia/geometry/calibration/pnp.py

```diff
@@ -0,0 +1,139 @@
+        raise AssertionError("world_points, img_points and intrinsics must have the same batch size.")
+
+    if world_points.shape[1] < 6:
+        raise AssertionError(
+            f"Atleast 6 points are required to use this function. "
+            f"Got {world_points.shape[1]} points."
+        )
+
+    # Checking if world_points (of any element of the batch) has rank = 3. This
+    # function cannot be used if all world_points (of any element of the batch) lie
+    # on a line or if all world points lie on a plane.
+    _, s, _ = torch.svd(world_points)
+    if torch.any(s[:, -1] < eps):
+        raise AssertionError(
+            f"The last singular value is smaller than {eps}. This function "
+            f"cannot be used if all world_points (of any element of the batch) "
+            f"lie on a line or if all world_points (of any element of the batch) "
+            f"lie on a plane."
+        )
+
+    B, N = world_points.shape[:2]
+    intrinsics_inv = torch.inverse(intrinsics)
+    world_points_h = convert_points_to_homogeneous(world_points)
+
+    # Getting normalized image points.
+    norm_points = kornia.geometry.transform_points(intrinsics_inv, img_points)
+
+    # Setting up the system (the matrix A in Ax=0)
+    system = torch.zeros((B, 2 * N, 12), dtype=world_points.dtype, device=world_points.device)
+    system[:, 0::2, 0:4] = world_points_h
+    system[:, 1::2, 4:8] = world_points_h
+    system[:, 0::2, 8:12] = world_points_h * (-1) * norm_points[..., 0:1]
+    system[:, 1::2, 8:12] = world_points_h * (-1) * norm_points[..., 1:2]
+
+    # Getting the solution vectors.
+    _, _, v = torch.svd(system)
+    solution = v[..., -1]
+
+    # Reshaping the solution vectors to the correct shape.
+    pred_world_to_cam = solution.reshape(B, 3, 4)
+
+    # We obtained one solution for each element of the batch. We may
+    # need to multiply each solution with a scalar. This is because
+    # if x is a solution to Ax=0, then cx is also a solution. We can
+    # find the required scalars by using the properties of
+    # rotation matrices. We do this in two parts:
+
+    # First, we fix the sign by making sure that the determinant of
+    # the all the rotation matrices are non negative (since determinant
+    # of a rotation matrix should be 1).
+    det = torch.det(pred_world_to_cam[:, :3, :3])
+    sign_fix = torch.where(det < 0, -1, 1)
+    pred_world_to_cam = pred_world_to_cam * sign_fix[:, None, None]
+
+    # Then, we make sure that norm of the 0th columns of the rotation
+    # matrices are 1. Do note that the norm of any column of a rotation
+    # matrix should be 1. Here we use the 0th column to calculate norm_col.
+    # We then multiply pred_world_to_cam with mul_factor.
+    norm_col = torch.sqrt(torch.sum(pred_world_to_cam[:, :3, 0] ** 2, axis=1))
+    mul_factor = (1 / norm_col)[:, None, None]
+    pred_world_to_cam = pred_world_to_cam * mul_factor
+
```

Comment:
could you add a TODO note here for the refinenement step ?

Reply:
I noticed that there are functions called `find_homography_dlt` and `find_homography_dlt_iterated`. Similarly, do you think having a separate `solve_pnp_dlt_iterated` instead of modifying `solve_pnp_dlt` would be nice ?

Also, do you think I should add a `NOTE` to the docstring of `solve_pnp_dlt` mentioning that the pose could be refined by using optimization or something? Or is just a TODO in the function to implement refinement in the future enough?

Reply:
add just todo